SWLOR.Game.Server/Feature/ShipModuleDefinition/CapitalWeaponsModuleDefinition.cs

```diff
@@ -0,0 +1,270 @@
+                        }
+                    }
+                    var defenderStat = GetAbilityScore(target, AbilityType.Vitality);
+                    var damage = Combat.CalculateDamage(
+                        attack,
+                        dmg,
+                        attackerStat,
+                        defense,
+                        defenderStat,
+                        0);
+
+                    var chanceToHit = Space.CalculateChanceToHit(activator, target);
+                    var roll = Random.D100(1);
+                    var isHit = roll <= chanceToHit;
+
+                    var sound = EffectVisualEffect(VisualEffect.Vfx_Ship_Blast);
+                    var missile = EffectVisualEffect(VisualEffect.Mirv_StarWars_Bolt2);
+
+                    if (attackType == "laser")
+                    {
+                        if (isHit)
+                        {
+                            AssignCommand(activator, () =>
+                            {
+                                ApplyEffectToObject(DurationType.Instant, sound, target);
+                                ApplyEffectToObject(DurationType.Instant, missile, target);
+
+                                DelayCommand(0.3f, () =>
+                                {
+                                    Space.ApplyShipDamage(activator, target, damage);
+                                });
+                            });
+                        }
+                        else
+                        {
+                            AssignCommand(activator, () =>
+                            {
+                                ApplyEffectToObject(DurationType.Instant, sound, target);
+                                ApplyEffectToObject(DurationType.Instant, missile, target);
+                            });
+                        }
+
+                        var attackId = isHit ? 1 : 4;
+                        var combatLogMessage = Combat.BuildCombatLogMessage(activator, target, attackId, chanceToHit);
+                        Messaging.SendMessageNearbyToPlayers(target, combatLogMessage, 60f);
+
+                        Enmity.ModifyEnmity(activator, target, damage);
+                        CombatPoint.AddCombatPoint(activator, target, SkillType.Piloting);
+                    }
+                    else if (attackType == "ion")
+                    {
+                        if (isHit)
+                        {
+                            AssignCommand(activator, () =>
+                            {
+                                var effect = EffectBeam(VisualEffect.Vfx_Beam_Mind, activator, BodyNode.Chest);
+                                ApplyEffectToObject(DurationType.Temporary, effect, target, 1.0f);
+                                ApplyEffectToObject(DurationType.Instant, EffectVisualEffect(VisualEffect.Vfx_Ion_Shot), activator);
+                            });
+
+                            DelayCommand(0.1f, () =>
```

Comment:
Another delay here which I'm not understanding. Unless you absolutely need it for some reason I would remove it.